hey-discount/src/main/webapp/resources/js/script.js

```diff
@@ -0,0 +1,878 @@
+        hashTags: ['hashtag'],
+        discount: '100',
+        validUntil: new Date('2021-07-06T23:00:00'),
+        rating: 0,
+        reviews: []
+    }
+
+];
+
+class App {
+    constructor() {
+        this.countOfPostsOnFeed = 0;
+        this.user = 'guest';
+        this.isVendor = false;
+        this.view = new View();
+        this.model = new Model();
+        this.model.restore();
+        if (!this.model.size()) {
+            this.model.addAll(ads);
+        }
+        this.filterConfig = {
+            vendor: null,
+            hashTags: null,
+            from: null,
+            to: null
+        };
+        this.adForEdit;
+        this.adForReview;
+        this.buttonLoadMore = document.getElementById("buttonMore");
+        this.buttonLoadMore.addEventListener("click", () => this.loadMoreAds());
+
+        this.buttonFilter = document.querySelector(".filter");
+        this.buttonFilter.addEventListener("submit", (event) => {
+            event.preventDefault();
+            this.filterAds(event);
+        });
+
+        document.getElementById("container").addEventListener("click", (event) => {
+            this.deleteOneAd(event);
+        });
+
+        document.getElementById("container").addEventListener("click", (event) => {
+            this.editOneAd(event);
+        });
+
+        document.getElementById("container").addEventListener("click", (event) => {
+            this.leaveReview(event);
+        });
+
+        this.buttonCancelReview = document.querySelector(".closeReview");
+        this.buttonCancelReview.addEventListener("click", () => this.cancelReview());
+
+        this.buttonPublishReview = document.getElementById("formLeavingReview");
+        this.buttonPublishReview.addEventListener("submit", (event) => {
+            event.preventDefault();
+            this.publishReview(event);
+        });
+
+        this.loadMoreAds();
+
+        this.buttonAddingNewAd = document.querySelector(".add-ad-button");
+        this.buttonAddingNewAd.addEventListener("click", () => this.addNewAd());
+
+        this.buttonCancelNewAd = document.querySelector(".closeAddAd");
+        this.buttonCancelNewAd.addEventListener("click", () => this.cancelNewAd());
+
+        this.buttonPublishAd = document.getElementById("formNewAd");
+        this.buttonPublishAd.addEventListener("submit", (event) => {
+            event.preventDefault();
+            this.publishAd(event);
+        });
+
+        this.buttonEnter = document.querySelector(".header-button");
+        this.buttonEnter.addEventListener("click", () => this.signUp());
+
+        this.buttonCancelEnter = document.querySelector(".closeSigningUp");
+        this.buttonCancelEnter.addEventListener("click", () => this.cancelSignUp());
+
+        this.buttonAuthorize = document.getElementById("formSigningUp");
+        this.buttonAuthorize.addEventListener("submit", (event) => {
+            event.preventDefault();
+            this.authorize(event);
+        });
```

Comment:
I'd advice you to move all event listeners code to a separate method (eg `assignEventListener`), because now constructor code is huge.